ecosystem/nft-metadata-crawler-parser/src/lib.rs

```diff
@@ -1,16 +1,26 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
+use anyhow::Context;
+use axum::response::Response;
+use bytes::Bytes;
+use reqwest::{header, Client};
+use std::{sync::Arc, time::Duration};
+use utils::constants::MAX_HEAD_REQUEST_RETRY_SECONDS;
+
+pub mod asset_uploader;
 pub mod config;
 pub mod models;
+pub mod parser;
 pub mod schema;
 pub mod utils;
 pub mod worker;
 
-use anyhow::Context;
-use reqwest::{header, Client};
-use std::time::Duration;
-use utils::constants::MAX_HEAD_REQUEST_RETRY_SECONDS;
+/// Trait for handling requests to the server
+#[async_trait::async_trait]
+trait Server: Send + Sync {
+    async fn handle_request(self: Arc<Self>, bytes: Bytes) -> anyhow::Result<Response<String>>;
+}
```

Comment:
I see here (https://github.com/aptos-labs/aptos-core/pull/13837#discussion_r1672731995) you chose not to implement `Json<Struct>` as the input type for the Axum server because of this trait. What is the point of this trait? Seems to just restrict you so you have to have this janky bytes thing. At the least make it take a `T` rather than `Bytes`, but idk what we get from this trait in the first place?

Reply:
Ah yeah ig for the trait I could just return a `Router` instead of a function that returns a `Response` haha, or just not use a trait at all

Reply:
```rust
trait Server: Send + Sync {
    fn build_router(self: Arc<Self>) -> Router;
}
```
😎